RULES/cc/cc.go

```diff
@@ -129,6 +157,20 @@ func (lib Library) Build(ctx core.Context) {
 	})
 }
 
+func (lib Library) WithToolchain(ctx core.Context, toolchain Toolchain) Library {
+	if !lib.multipleToolchains {
+		return lib
```

Comment:
yeah this was a bit of a hack to let stddeps compile with their own library (otherwise they'd be compiled with the same library that requires them, which is bad). I think I have a better solution tho

Reply:
What I said before is not correct (especially the stddeps thing is just wrong). I do have a reason for this. The way I think about it is that a Library without `multipleToolchains` is one that, irrespective of who is depends on it, is always compiled with the same `lib.Toolchain`. It's not one that fails when someone with another toolchain depends on it. 

Let me know if that sounds too strange. 

Ultimately (for me) I think it's down to preference, and the reason I have this is for convenience when defining libsupcxx custom toolchains. If we decide we don't like this approach I need to rethink how I do things there a little bit.

Reply:
I think this is ok, but then I think we should call the "constructor" `ToolchainAware` or sth similar. That way you can have `cc.Library{...}.ToolchainAware()` to get a toolchain-aware library. I think that is clearer (form a user perspective) than `MultipleToolchains`.

Reply:
I found a better solution for libsupcxx's boot libraries (which were the reason for this). I made a new commit to the PR over there.
We can change this so that it's an error when you try to `WithToolchain` on an unsupported toolchain. It's really down to preference now, libsupcxx is unaffected.